yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -51,6 +51,9 @@ private[spark] abstract class YarnSchedulerBackend(
 
   private implicit val askTimeout = RpcUtils.askRpcTimeout(sc.conf)
 
+  // Flag to specify whether AM is registered or not.
+  @volatile private var isAmRegistered = false
```

Comment:
Sure, I will fix it. From my guessing, I think it should be OK to call reset on the first registration.